wadas/ui/ai_model_download_dialog.py

```diff
@@ -238,23 +224,35 @@ def get_selected_models_by_type(self):
             return classification_models, detection_models
 
         for i in range(layout.count()):
-            scroll_area = layout.itemAt(i).widget()
-            if isinstance(scroll_area, QScrollArea):
-                group_box = scroll_area.widget()
-                if isinstance(group_box, QGroupBox):
-                    group_title = group_box.title().lower()
-                    inner_layout = group_box.layout()
-                    if inner_layout is None:
-                        continue
-
-                    for j in range(inner_layout.count()):
-                        checkbox = inner_layout.itemAt(j).widget()
-                        if isinstance(checkbox, QCheckBox) and checkbox.isChecked():
-                            model_name = checkbox.text().replace(" (default)", "")
-                            if "classification" in group_title:
-                                classification_models.append(model_name)
-                            elif "detection" in group_title:
-                                detection_models.append(model_name)
+            groupbox = layout.itemAt(i).widget()
+            if not isinstance(groupbox, QGroupBox):
+                continue
+
+            group_title = groupbox.title().lower()
+            groupbox_layout = groupbox.layout()
+            if groupbox_layout is None or groupbox_layout.count() == 0:
+                continue
```

Comment:
if (groupbox_layout := groupbox.layout()) is None or not groupbox_layout.count():
                continue